### PR TITLE
Check nullable to detect null type in 3.1 spec

### LINKED ISSUE #18184: [BUG][multiple][OpenAPI-v3.1.0] SIMPLIFY_ONEOF_ANYOF is broken in v7.3.0,v7.4.0
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

On v.7.3.0 (and  v7.4.0, the current latest release) the **simplify-oneof-anyof** normalizer was broken (at least for Python, but I believe for other generators as well). For optional model fields, an additional model is created which makes usage very cumbersome.

For example, consider an `Item` model with the following optional fields:
1. `my_enum` - optional enum of type `MyEnum`
2. `is_offer` - optional `bool`

The expected behavior is something like:
```python
class Item(BaseModel):
    is_offer: Optional[StrictBool] = None
    my_enum: Optional[MyEnum] = None
    ...
```

But  the actual behavior in the specified generator versions produced additional models and `Item` looks like:
```python
class Item(BaseModel):
    is_offer: Optional[IsOffer] = None
    my_enum: Optional[ItemMyEnum] = None
    ...
```

##### openapi-generator version

Bug appeared in v7.3.0, still in v7.4.0 and latest `master` (and didn't not exist in v7.2.0).

##### OpenAPI declaration file content or url

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Generation Details

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Steps to reproduce

See full repro details in [gist](https://gist.github.com/guyarad/21bdd60b5643ac0d13ae5a57c593eefc#file-simplify_oneof_anyof-bug-md)

##### Suggest a fix

**TL;DR** - the offending commit seems to be [this](https://github.com/OpenAPITools/openapi-generator/commit/301f25a3068db6dd17ab930f72975b235664f172) one. It changes the way `null` schema is identified, causing the simplification normalizer failing to do its job.

After cloning and using the "docker development" route I was able to use `git bisect` and each building+generating the client, identifying the commit quickly, so thank you for offering the docker packaging!

I believe the fix should include added tests for this behavior (not sure why existing tests did not identify it).

### PR DESCRIPTION
**Pull request by wing328**
- Fix #18184
- Add a test

fyi @OpenAPITools/generator-core-team 

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java [hunk 1]
```diff
@@ -513,4 +513,23 @@ public void testSetContainerToNullable() {
         assertEquals(((Schema) schema4.getProperties().get("set_property")).getNullable(), true);
         assertEquals(((Schema) schema4.getProperties().get("map_property")).getNullable(), null);
     }
+
+    @Test
```
Comment (guyarad) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java:517]: why would it be different (in terms of what we are testing) than `testOpenAPINormalizerSimplifyOneOfAnyOf`?
and `testOpenAPINormalizerSimplifyOneOfAnyOfStringAndEnumString`

why not have same implementation, but different source spec?

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java [hunk 2]
```diff
@@ -513,4 +513,23 @@ public void testSetContainerToNullable() {
...
+
+        Map<String, String> options = new HashMap<>();
+        options.put("SIMPLIFY_ANYOF_STRING_AND_ENUM_STRING", "true");
+        OpenAPINormalizer openAPINormalizer = new OpenAPINormalizer(openAPI, options);
+        openAPINormalizer.normalize();
+
+        Schema schema2 = openAPI.getComponents().getSchemas().get("Item");
+        assertEquals(((Schema) schema2.getProperties().get("my_enum")).getAnyOf(), null);
+        assertEquals(((Schema) schema2.getProperties().get("my_enum")).get$ref(), "#/components/schemas/MyEnum");
+    }
```
Comment (guyarad) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java:534]: note that the generation created models for the boolean field (which you didn't take) and the query string `Q`. consider testing those too.

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java:534]: sure. can you file a PR to add those tests after this PR is merged?

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java:517]: sure, not against it. can you file a PR to add those tests after this PR is merged?

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/OpenAPINormalizerTest.java:517]: we will get it merged first and then create an issue to track better openapi spec 3.1 test

**Pull request merged**